lib/index.js

```diff
@@ -1167,17 +1168,38 @@ KnexMigrator.prototype.rollback = function rollback(options) {
 
             resolve();
         }).then(function () {
-            return self.connection('migrations')
-                .where({
+            let whereQuery = {};
+
+            // CASE 1: rollback to specific version
+            // CASE 2: rollback current version you are on (query all and filter out)
+            if (version) {
+                debug(`Rollback to specific version: ${version}`);
+                whereQuery = {};
+            } else {
+                whereQuery = {
                     currentVersion: self.currentVersion
-                })
+                };
+            }
+
+            return self.connection('migrations')
+                .where(whereQuery)
                 .then(function (values) {
                     if (!values.length) {
                         throw new errors.IncorrectUsageError({
                             message: 'No migrations available to rollback.'
                         });
                     }
 
+                    // CASE: filter out all versions which are smaller than the version we want to rollback to
+                    if (version) {
+                        values = _.filter(values, function (value) {
+                            return utils.isGreaterThanVersion({
+                                greaterVersion: value.version,
+                                smallerVersion: version
```

Comment:
We should only use major and minor versions here. You could rollback to e.g. 1.25.3.

Reply:
Hm no looks fine.